### ISSUE TITLE
Extending recipe.json style categorization

### ISSUE THREAD

**Issue by pricelessbrewing** [beerjson/beerjson#38]
Currently it's limited to. 
["lager", "ale", "mead", "wheat", "mixed", "cider"]

Again with the "wheat" thing that IMO makes no sense. What is that, a subcategory of ale/lager, any beer recipe including wheat? Not a fan. 

I'm going to propose the following. Does that seem sufficient, any other cases I'm missing? 

["ale", "brett", "cider", "kombucha", "lager", "mead", "mixed", "soda", "sour", "spontaneous", "wine"]

Additionally, I don't think this should be a strict categorization, if so we should allow an "other" category in case something isn't supported properly. 

I'll be submitting a push in the next day or two.

**Comment by rmsrosa** [comment]
I like that. 

The original classification already seemed a mixed bag, not a very precise categorization. Mead and Cider are at the same level of Beer, as types of beverages, not really characteristic fermentations. Wheat is also an odd item in the bag, as you mentioned. So I think we should make it clearly flexible, allowing for "other". 

But I also think we should add "beer". BeerXML specified the type of beer (ale, lager, wheat), but  BJCP itself does not specify it. In fact, some styles can be brewed with either ale or lager yeast, such as Baltic Porter, American Wheat Beer, Bière de Garde, and Historical Beer: Pre-Prohibition Porter. So, if a style is clearly an ale or a lager, someone may classified it as so, otherwise, we should allow for "beer". And, as I said above, "beer" is at the same level of "mead" and "cider", so this is another reason to include it.

And other classifications, such as that from the Brewers Association, are not that precise, either. So, I agree with your proposal of extending the classifications, with the choice "other", but I would also include "beer":

["ale", "beer", "brett", "cider", "kombucha", "lager", "mead", "mixed", "other", "soda", "sour", "spontaneous", "wine"]

**Comment by pricelessbrewing** [comment]
Hmm makes sense, I was thinking from a recipe standpoint, and not from a style persepective. 

Obviously whatever recipe you're working from will either be a lager, or ale for those styles, but the style standard itself can be either for the styles you listed (as well as a few others). 

This is interesting, do we A) Try and include all types of fermented drink types? 

["ale", "beer", "brett", "cider", "kombucha", "lager", "mead", "mixed", "other", "soda", "sour", "spontaneous", "wine"]

But then we get into things like, cyser, braggot, pyment etc which are between cider and mead, beer and mead, and wine and mead. Or the ale OR lager styles as mentioned above. 

or B) Keep it to the main types,

["beer", "cider", "kombucha", "mead", "other", "soda", "wine"]. 

Then allow the style description to designate the sub category? 

Thoughts @krutilin @superroma @rocketman768 ?

**Comment by rocketman768** [comment]
Thanks for the discussion, and I think the issue is valid and the categories do deserve some change. I think in the end, it comes down to what feature you want to enable. For me personally, the BeerXML equivalent was not so useful. Maybe other people are making lots of different beverages and really want to sort based on all these types though?

A lot of the proposed types overlap with the type of yeast or bacteria in the recipe (ale, brett, lager, sour), so it seems I can just sort recipes by yeast type already. What's left seems to be related to the major ingredients, which is probably more useful and not so easily derived.

I don't know what the categories should be, but I think it's worth thinking about what kind of functionality you want to enable and what kind of brewer you are targeting (sake?). As far as the BJCP is concerned, everything they cover is just "beer," "mead," or "cider."

**Comment by krutilin** [comment]
Meanwhile, I added the test for the BJCP 2015 styles converted to the current beerjson format by @rmsrosa. See following branch https://github.com/beerjson/beerjson/tree/BJCP-styles. The test successfully failed, because there is no such type as "beer" in our style schema.

I like the idea to extend StyleCategories and include all types of fermented drink types. This will give the answer to the brewer: What kind of drink I will brew according to this recipe as a result, after completing all brewing steps? 

Binding to the yeast or bacteria in the recipe is the answer to the question "Wich" yeast or bacteria I should use in this concrete recipe for my mash without information about previous steps. As @rocketman768 mentioned.

Binding to the style is more compete.

**Comment by rmsrosa** [comment]
I was initially inclined to option (A) of @pricelessbrewing, mostly for compatibility with the previous definition in BeerXML, but we all seem to agree that keeping to the main types of fermentable drinks makes more sense, so option (B) of @pricelessbrewing seems better indeed. Let's go with that.

**Comment by jdominiczak** [comment]
I agree with option (B) from @pricelessbrewing.  I think that there is quite a bit of unnecessary complexity to option (A) and the fact that there are some crossover beverages that could be described as up to three of the options in A.  I think the simpler option with a option for "Other" would be best.

I also think this is the point of a style selection, to define what is being made with an eye to more detail.

**Comment by krutilin** [comment]
Review my PR, I think it should solve the issue.

**Comment by rmsrosa** [comment]
In the first json file for the BJCP2015 style guide that I shared in the gitter chat, I mistakenly labeled the value for the color as "density", instead of "scale". I fixed that, although I would still prefer to standardize all of them to "value", regardless of the unit, and use "unit" instead of "units". But I will leave it as specified in the current beerjson format. If we change it, I will be happy to modify the json guide and send it again. Here is the link for the corrected file:

https://www.dropbox.com/s/8omfpj9qyh2qjqo/BJCP2015styleguide.json?dl=0
